### PR TITLE
refactor: refresh infligh queue if received a REGACK

### PR DESCRIPTION
**Pull request by HJianBo**

### CODE REVIEW

File: src/emqx_sn_frame.erl [hunk 1]
```diff
@@ -268,40 +268,81 @@ message_type(16#1d) ->
 message_type(Type) ->
     io_lib:format("Unknown Type ~p", [Type]).
 
+format(?SN_CONNECT_MSG(Flags, ProtocolId, Duration, ClientId)) ->
+    #mqtt_sn_flags{
+       will = Will,
+       clean_start = CleanStart} = Flags,
+    io_lib:format("SN_CONNECT(W~w, C~w, ProtocolId=~w, Duration=~w, "
+                  "ClientId=~s)",
```
Comment (zmstone) [inline src/emqx_sn_frame.erl:276]: is `ClientId` always a printable string ?
should it be `~ts` instead ?

File: src/emqx_sn_gateway.erl [hunk 2]
```diff
@@ -729,8 +730,17 @@ mqtt2sn(?PUBCOMP_PACKET(MsgId), _State) ->
...
+  #mqtt_packet{header = #mqtt_packet_header{
+                           type = ?PUBLISH,
+                           qos = QoS,
+                           %dup = Dup,
+                           retain = Retain},
+               variable = #mqtt_packet_publish{
+                             topic_name = Topic,
+                             packet_id = PacketId},
+               payload = Payload}, #state{clientid = ClientId}) ->
+    NPacketId = if QoS =:= ?QOS_0 -> 0;
```
Comment (zmstone) [inline src/emqx_sn_gateway.erl:743]: nit: use `case`

File: src/emqx_sn_gateway.erl [hunk 3]
```diff
@@ -771,9 +784,10 @@ send_connack(State) ->
 
 send_message(Msg = #mqtt_sn_message{type = Type},
              State = #state{sockpid = SockPid, peername = Peername}) ->
-    ?LOG(debug, "SEND ~s~n", [emqx_sn_frame:format(Msg)]),
+    ?LOG(info, "SEND ~s~n", [emqx_sn_frame:format(Msg)]),
```
Comment (zmstone) [inline src/emqx_sn_gateway.erl:787]: is this really intended change ?

File: src/emqx_sn_gateway.erl [hunk 4]
```diff
@@ -771,9 +784,10 @@ send_connack(State) ->
...
     inc_outgoing_stats(Type),
     Data = emqx_sn_frame:serialize(Msg),
+    ?LOG(debug, "SEND ~0p", [Data]),
```
Comment (zmstone) [inline src/emqx_sn_gateway.erl:790]: should we really log twice for the same message ?

Comment (HJianBo) [reply src/emqx_sn_gateway.erl:791]: Does this look better? Under the info level, it is possible to print the content of the message in a human-readable way. Printing out binary data messages again at debug level would help diagnose problems with message serialization?

Comment (zmstone) [reply src/emqx_sn_gateway.erl:791]: I’d argue both should be debug

**Review by zmstone — APPROVED** [review]

File: src/emqx_sn_gateway.erl (hunk 4 above)
Comment (HJianBo) [reply src/emqx_sn_gateway.erl:791]: fixed

**Pull request merged**